workorders.lic

```diff
@@ -172,48 +187,148 @@ class WorkOrders
     fput('go door')
   end
 
+  def count_combine_bone(stock_room, bone, bone_stock, quantity, volume)
+    echo("In count-combine-bone")
+    if inside?("first #{bone} stack", @bag) && inside?("second #{bone} stack", @bag)
+      bput("get #{bone} stack from my #{@bag}", 'You get')
+      bput("get #{bone} stack from my #{@bag}", 'You get')
+      case bput('combine', 'You combine', 'The resulting bone stack')
+      when 'The resulting bone stack'
+        stow_hands
+      when 'You combine'
+        case bput("get #{bone} stack from my #{@bag}", 'You get', 'What were')
+        when 'You get'
+          case bput('combine', 'You combine', 'The resulting bone stack')
+          when 'The resulting bone stack', 'You combine'
+            stow_hands
+          end
+        end
+      end
+#    else
+#      order_item(stock-room, bone-stock)
+    end
+    if inside?("first #{bone} stack", @bag)
+      bone_have = bput("count #{bone} stack", 'You count out \d+ pieces').scan(/\d+/).first.to_i
+      echo ("bone_have is #{bone_have}")
+      pause 3
+      bone_recipe = (quantity * volume)
+      bone_need = ((bone_recipe - bone_have) / 10.0).ceil
+      echo("bone need is #{bone_need}")
+      if bone_have <= bone_recipe
+         bone_need.times do
+           order_item(stock_room, bone_stock)
+           next unless left_hand && right_hand
+           bput('combine', 'You combine')
+         end
+      end      
+      bone_need.times do
+        bput("get #{bone} stack from my #{@bag}", 'You get', 'What were')
+        bput("get #{bone} stack from my #{@bag}", 'You get', 'What were')
+        bput('combine', 'You combine', 'The resulting bone stack', 'You must be')
+      end
+#    end
+      stow_hands
+    else
+      bone_need = ((quantity * volume) / 10.0).ceil
+      echo("bone needed = #{bone_need}")
+      pause 3
+      bone_need.times do
+        order_item(stock_room, bone_stock)
+        next unless left_hand && right_hand
+        bput('combine', 'You combine')
+      end
+      bone_need.times do
```

Comment:
How does this work? Won't all of the bone stacks be combined from line 238?

You get twice, combine, which leaves one item in your hands, then you would get twice on the next loop. If this executes more than once, won't you get an error on that second loop?